src/auth/flows.rs

```diff
@@ -887,3 +896,155 @@ pub async fn auth_callback(
         Err(AuthenticationError::InvalidCredentials)
     }
 }
+
+#[derive(Deserialize, Validate)]
+pub struct NewAccount {
+    #[validate(length(min = 1, max = 39), regex = "RE_URL_SAFE")]
+    pub username: String,
+    #[validate(length(min = 8, max = 256))]
+    pub password: String,
+    #[validate(email)]
+    pub email: String,
+    pub challenge: String,
+}
+
+#[post("create")]
+pub async fn create_account_with_password(
+    req: HttpRequest,
+    pool: Data<PgPool>,
+    redis: Data<deadpool_redis::Pool>,
+    new_account: web::Json<NewAccount>,
+) -> Result<HttpResponse, ApiError> {
+    new_account
+        .0
+        .validate()
+        .map_err(|err| ApiError::InvalidInput(validation_errors_to_string(err, None)))?;
+
+    if check_turnstile_captcha(&req, &new_account.challenge).await? {
+        return Err(ApiError::Turnstile);
+    }
+
+    if crate::database::models::User::get(&new_account.username, &**pool, &redis)
```

Comment:
We should also do an email collision check here- no duplicate emails (at least against verified emails?))